posts/espp-in-this-economy/index.qmd

```diff
@@ -0,0 +1,215 @@
+  step: 0.5,
+  value: 100,
+});
+
+viewof lookback_price_at_purchase = Inputs.number([0, Infinity], {
+  label: "Stock Price (purchase date)",
+  type: "number",
+  step: 0.5,
+  value: 105,
+});
+
+
+html `<hr>`;
+
+{
+  if (lookback_price_at_start < lookback_price_at_purchase) {
+    return html `The stock <strong>rose</strong> in value.<hr>The ${lookback_discount_percent}% discount is applied to the <strong>Stock Price (start)</strong> price.`;
+  } else {
+    return html `The stock <strong>fell</strong> in value.<hr>The ${lookback_discount_percent}% discount is applied to the <strong>Stock Price (purchase date)</strong> price.`;
+  }
+}
+
+lookback_purchase_price_with_discount = (1.0 - (lookback_discount_percent / 100.0)) * Math.min(lookback_price_at_start, lookback_price_at_purchase);
+html `<hr>`;
+html `Each share is purchased for <strong>$${lookback_purchase_price_with_discount.toFixed(2)}</strong> and is immediately worth <strong>$${lookback_price_at_purchase.toFixed(2)}</strong>`;
+
+lookback_average_escrow_time_in_months = lookback_purchase_period_in_months / 2.0;
+lookback_average_escrow_time_in_years = lookback_average_escrow_time_in_months / 12.0;
+lookback_interest_rate = Math.pow(lookback_price_at_purchase / lookback_purchase_price_with_discount, 1.0 / lookback_average_escrow_time_in_years) - 1.0;
+
+html `<hr>`;
+html `Annualized interest rate: <strong class="fs-1">${(100.0 * lookback_interest_rate).toFixed(2)}%</strong>`;
+```
+
+## Caveats
+
+ESPP administration and tax law have caveats that change the math in some scenarios. However, I have never seen these issues alone make a plan undesirable.
+
+Capturing the exceptional interest rate requires selling the shares as quickly as possible every time. I set a recurring reminder for myself so I do not forget.
+
+Commissions and fees reduce the effective interest rate slightly.
+
+Some plans allow purchasing partial shares. Others may only allow whole shares; in that case, the "change" will be refunded on a future paycheck.
+
+Qualified (tax efficient) plans under [&sect;423](https://www.law.cornell.edu/uscode/text/26/423) are limited to $25,000 of stock per year. With a 15% discount, the maximum contribution works out to be $21,250.
+
+When a stock rises in value over the offering period, plans with a lookback feature may limit the number of shares that can be purchased. The "change" above and beyond this amount will again be refunded, but it will also lower the effective interest rate.
+
+If an employee is terminated (voluntarily or involuntarily), money contributed during a purchase period will be refunded on their final paycheck or shortly after.
+
+Some plans allow employees to withdraw from the plan during a purchase period and have the money refunded on a future paycheck. There may be significant delay between withdrawing from the plan and receiving the money.
+
+## Planning
+
+How does an ESPP fit into a tax-efficient financial plan? It definitely depends.
```

Comment:
Maybe this would be the spot to bring up the deferred income question. "It definitely depends. Contributing to ESPP does mean you are deferring a portion of your income for a few months. If you have outstanding debts or are saving toward other short-term goals, this may not be right for you. But if you can hold off on realizing that salary for just a short while, you stand to reap the financial gains."

Reply:
This is a good nudge. I will make expand it a bit.